src/org/labkey/test/components/ManageSampleStatusesPanel.java

```diff
@@ -0,0 +1,188 @@
+        status.statusType = getStatusType();
+        status.label = getLabel();
+        status.description = getDescription();
+
+        return status;
+    }
+
+    public String getLabel()
+    {
+        return elementCache().labelField.getValue();
+    }
+
+    public String getDescription()
+    {
+        return elementCache().descriptionField.getText();
+    }
+
+    public SampleTypeHelper.StatusType getStatusType()
+    {
+        return SampleTypeHelper.StatusType.valueOf(elementCache().statusTypeSelect.getValue());
+    }
+
+    public List<String> getStatusNames()
+    {
+        return elementCache().statusItems
+                .findElements(this)
+                .stream()
+                .map(WebElement::getText)
+                .collect(Collectors.toList());
+    }
+
+    public ManageSampleStatusesPanel clickAddStatus()
+    {
+        elementCache().addStatusButton.click();
+        return this;
+    }
+
+    public ManageSampleStatusesPanel setLabel(String label)
+    {
+        elementCache().labelField.setValue(label);
+        return this;
+    }
+
+    public ManageSampleStatusesPanel setDescription(String description)
+    {
+        elementCache().descriptionField.clear();
+        if (description != null)
+        {
+           elementCache().descriptionField.sendKeys(description);
+        }
+        return this;
+    }
+
+    public ManageSampleStatusesPanel setStatusType(SampleTypeHelper.StatusType statusType)
+    {
+        elementCache().statusTypeSelect.select(statusType.toString());
+        return this;
+    }
+
+    public ManageSampleStatusesPanel clickSave()
+    {
+        elementCache().saveButton.click();
+        return this;
+    }
+
+    public ManageSampleStatusesPanel addStatus(String label, String description, SampleTypeHelper.StatusType statusType)
+    {
+        clickAddStatus()
+                .setLabel(label)
+                .setDescription(description)
+                .setStatusType(statusType);
+        elementCache().saveButton.click();
+        return this;
+    }
+
+    public ManageSampleStatusesPanel deleteStatus(String label)
+    {
+
+        elementCache().deleteButton.click();
+        return this;
+    }
+
+    protected class ElementCache extends Component.ElementCache
+    {
+        WebElement statusItem(String name, SampleTypeHelper.StatusType statusType)
+        {
+            if (statusType == null || statusType.toString().equals(name))
+                return Locator.tagWithClass("button", "list-group-item").containing(name).findElement(getComponentElement());
+            else
+                return Locator.tagWithClass("button", "list-group-item").containing(name +  statusType).findElement(getComponentElement());
+        }
+
+        Locator statusItems = Locator.tagWithClass("button", "list-group-item");
+
+        WebElement addStatusButton = Locator.tagWithText("span", "Add New Status")
+                .findWhenNeeded(getComponentElement());
```

Comment:
I _believe_  `this` and `getComponentElement` are equivalent.

Reply:
Leaving as is since it seems to work.